### PR TITLE
Minor type utils code refactoring

### PR DESCRIPTION
**Pull request by smallx**
no RN/DOCS

### CODE REVIEW

File: core/trino-spi/src/main/java/io/trino/spi/type/TypeUtils.java [hunk 1]
```diff
@@ -69,30 +69,33 @@ public static void writeNativeValue(Type type, BlockBuilder blockBuilder, @Nulla
...
-            type.writeDouble(blockBuilder, ((double) value));
-        }
-        else if (type.getJavaType() == long.class) {
-            type.writeLong(blockBuilder, ((long) value));
-        }
-        else if (type.getJavaType() == Slice.class) {
-            Slice slice;
-            if (value instanceof byte[]) {
-                slice = Slices.wrappedBuffer((byte[]) value);
+        else {
```
Comment (raunaqmorarka) [inline core/trino-spi/src/main/java/io/trino/spi/type/TypeUtils.java:72]: you can unwrap this `else` by adding a return statement in the `if` branch

**Review by raunaqmorarka — COMMENTED** [review]
Could you provide some motivation for this change ?
I'm not sure that this actually changes performance or readability of this code.

**Comment by smallx** [comment]
> Could you provide some motivation for this change ? I'm not sure that this actually changes performance or readability of this code.

@raunaqmorarka These are the changes:
* `(Boolean) value` -> `(boolean) value`
* `((double) value)` -> `(double) value`
* `((long) value)` -> `(long) value`
* multi `type.getJavaType()` -> save to variable, as `type.getJavaType()` is virtual function call.

File: core/trino-spi/src/main/java/io/trino/spi/type/TypeUtils.java (hunk 1 above)
Comment (smallx) [reply core/trino-spi/src/main/java/io/trino/spi/type/TypeUtils.java]: updated

**Comment by smallx** [comment]
I have already sent the CLA :)

**Comment by raunaqmorarka** [comment]
> > Could you provide some motivation for this change ? I'm not sure that this actually changes performance or readability of this code.
> 
> @raunaqmorarka These are the changes:
> 
> * `(Boolean) value` -> `(boolean) value`
> * `((double) value)` -> `(double) value`
> * `((long) value)` -> `(long) value`
> * multi `type.getJavaType()` -> save to variable, as `type.getJavaType()` is virtual function call.

It's not clear that this would change anything for performance, after JIT runs the code won't necessarily be executed the way it is written (e.g. the if-else ladder might be optimised into a switch case). Could you try a small JMH benchmark to verify if this has any implication for performance ?

**Comment by smallx** [comment]
Thanks everyone. I don't think this pull request is very helpful, so I will close it.

**Pull request closed without merging**